src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```
@@ -0,0 +1,143 @@
+                            final SimpleCmdExecutor simpleCmdExecutor,
+                            final GridEngineCommandCompiler commandCompiler) {
+        this.jobMapper = jobMapper;
+        this.simpleCmdExecutor = simpleCmdExecutor;
+        this.commandCompiler = commandCompiler;
+    }
+
+    @Override
+    public EngineType getProviderType() {
+        return EngineType.SLURM;
+    }
+
+    @Override
+    public Listing<Job> filterJobs(final JobFilter jobFilter) {
+        final CommandResult result = simpleCmdExecutor.execute(makeSqueueCommand(jobFilter));
+        if (result.getExitCode() != 0) {
+            CommandsUtils.throwExecutionDetails(result);
+        } else if (!result.getStdErr().isEmpty()) {
+            log.warn(result.getStdErr().toString());
+        }
+        return mapToJobListing(result.getStdOut());
+    }
+
+    /**
+     * Creates the structure of an executable command based on the passed filter.
+     *
+     * @param filter User-defined filter for selecting job data.
+     * @return The structure of an executable command.
+     */
+    private String[] makeSqueueCommand(final JobFilter filter) {
+        final Context context = new Context();
+        context.setVariable(JOB_FILTER, filter);
+        return commandCompiler.compileCommand(getProviderType(), SQUEUE_COMMAND, context);
+    }
+
+    private Listing<Job> mapToJobListing(final List<String> stdOut) {
+        if (stdOut.size() > 1) {
+            return new Listing<>(stdOut.stream()
+                    .skip(1)
+                    .map(SacctCommandParser::parseSlurmJob)
+                    .filter(s -> !s.isEmpty())
+                    .map(SacctCommandParser::mapJobDataToSlurmJob)
+                    .map(jobMapper::slurmJobToJob)
+                    .collect(Collectors.toList()));
+        }
+        return new Listing<>();
+    }
+
+    @Override
+    public Job runJob(final JobOptions options) {
+        return null;
```

It is better to throw `UnsupportedOperationException` instead of returning `null` for methods, that are going to implement later.

---

done